components/brave_wallet_ui/common/slices/endpoints/off-ramp.endpoints.ts

```diff
@@ -0,0 +1,103 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Types
+import {
+  BraveWallet,
+  SupportedOffRampNetworks
+} from '../../../constants/types'
+import {
+  WalletApiEndpointBuilderParams
+} from '../api-base.slice'
+
+// Utils
+import {
+  getBatTokensFromList,
+  getNativeTokensFromList,
+  getUniqueAssets
+} from '../../../utils/asset-utils'
+import { addLogoToToken } from '../../async/lib'
+import { mapLimit } from 'async'
+
+export const offRampEndpoints = ({
+  query
+}: WalletApiEndpointBuilderParams) => {
+  return {
+    getOffRampAssets: query<
+      {
+        rampAssetOptions: BraveWallet.BlockchainToken[]
+        allAssetOptions: BraveWallet.BlockchainToken[]
+      },
+      void
+    >({
+      queryFn: async (_arg, _store, _extraOptions, baseQuery) => {
+        try {
+          const {
+            data: { blockchainRegistry }
+          } = baseQuery(undefined)
+          const { kRamp } = BraveWallet.OffRampProvider
+
+          const rampAssets = await mapLimit(
+            SupportedOffRampNetworks,
+            10,
+            async (chainId: string) =>
+              await blockchainRegistry.getSellTokens(kRamp, chainId)
+          )
+
+          // add token logos
+          const rampAssetOptions: BraveWallet.BlockchainToken[] =
+            await mapLimit(
+              rampAssets.flatMap((p) => p.tokens),
+              10,
+              async (token: BraveWallet.BlockchainToken) =>
+                await addLogoToToken(token)
+            )
+
+          // separate native assets from tokens
+          const {
+            tokens: rampTokenOptions,
+            nativeAssets: rampNativeAssetOptions
+          } = getNativeTokensFromList(rampAssetOptions)
+
+          // separate BAT from other tokens
+          const {
+            bat: rampBatTokens,
+            nonBat: rampNonBatTokens
+          } = getBatTokensFromList(rampTokenOptions)
+
+          // moves Gas coins and BAT to front of list
+          const sortedRampOptions =
+            [
+              ...rampNativeAssetOptions,
+              ...rampBatTokens,
+              ...rampNonBatTokens
+            ]
+
+          const results = {
+            rampAssetOptions: sortedRampOptions,
+            allAssetOptions: getUniqueAssets([
+              ...sortedRampOptions
+            ])
+          }
+
+          return {
+            data: results
+          }
+        } catch (error) {
+          const errorMessage = `Unable to fetch offRamp assets: ${error}`
+          console.log(errorMessage)
+          return {
+            error: errorMessage
+          }
+        }
+      }, providesTags: (results, error, _arg) => {
+        if (error) {
+          return ['UNKNOWN_ERROR']
+        }
+        return ['OffRampAssets']
+      }
+    })
+  }
+}
```

Comment:
nit: Missing new line at end of file

Reply:
It actually is there `Github` just doesn't show it 

![Screenshot 39](https://github.com/brave/brave-core/assets/40611140/87f6a742-c233-4058-95ad-2e16b0f8aca4)

If it were missing `Github` would display an error like so

![Screenshot 42](https://github.com/brave/brave-core/assets/40611140/3fdf6ebf-eed9-4252-8ebc-ba9249223bfa)